### PR TITLE
python: Use Python's logging through pyo3_log

### PR DESCRIPTION
**Pull request by chrysn**
This makes logging accessible and configurable through Python.

### CODE REVIEW

**Comment by chrysn** [comment]
I've used this a lot with own added info messages; the trace messages emitted by lakers itself are not visible in this unless we touch [one more knob](https://docs.rs/pyo3-log/latest/pyo3_log/struct.Logger.html#method.filter) -- might make sense to turn that on by default, because nothing we do is performance critical on the Python side.

File: lakers-python/src/lib.rs [hunk 1]
```diff
@@ -72,8 +71,10 @@ impl AutoCredential {
 #[pyo3(name = "lakers")]
 fn lakers_python(_py: Python, m: &Bound<'_, PyModule>) -> PyResult<()> {
     // initialize the logger once when the module is imported
```
Comment (geonnave) [inline lakers-python/src/lib.rs:73]:
```
// enables `logging.basicConfig(...)` from python and then logs from this wrapper just work
```

**Comment by geonnave** [comment]
Cool! Suggested a comment just for clarity.

Could you enable that knob so that we don't lose the from-Rust log functionality that we currently have? Is it possible to have it controlled by an env var like with env_logger?

**Comment by chrysn** [comment]
I'll add some API expositure and some basicConfig reference.

**Comment by chrysn** [comment]
I misunderstood how this can be configured – more API won't do us any good. Instead, given that we're not logging a whole lot, and are not performance critical in Python, I've changed the code to unconditionally log trace from Rust. (This is still not super slow, as all pyo3_log's caching etc is still in). There are module level docs now that explain the logging isutation.

File: lakers-python/src/lib.rs [hunk 2]
```diff
@@ -67,13 +66,26 @@ impl AutoCredential {
...
 
-// this name must match `lib.name` in `Cargo.toml`
+/// Lakers implementation of EDHOC.
+///
+/// The `EdhocInitiator` and `EdhocResponder` are entry points to this module.
+///
+/// Operations in this module produce logging entries on the `lakers.initiator` and
+/// `lakers.responder` logger names. Due to implementation details of `pyo3_log`, Python's log
+/// levels are cached in the Rust implementation. It is recommended that the full logging
+/// is configured before creating Lakers objects. A setup with `logging.basicConfig(loglevel=5)`
```
Comment (geonnave) [inline lakers-python/src/lib.rs:76]: Maybe best to add log level name? like `logging.basicConfig(level=logging.DEBUG)`

Comment (geonnave) [reply lakers-python/src/lib.rs:76]: Actually I am still confused on how to use it. 

I tried:

```python
import lakers, logging
logging.basicConfig(level=logging.DEBUG)
lakers.EdhocInitiator()
```

but after `python3 testlog.py` nothing got emitted.

So maybe I misunderstood the part about the hardcoded logs from Rust?

Comment (geonnave) [reply lakers-python/src/lib.rs:76]: I am also not sure it is a good idea to have Rust always-logging by default. I guess I could live with it sending only trace logs.
On the other hand, if we add more logs to the Rust side, we should make sure the python wrapper sets the Rust level to trace during build.

Comment (chrysn) [reply lakers-python/src/lib.rs:76]: So, the thing is that Python has no level below DEBUG (which Rust's log knows as 'trace'). That's why the docs say `basicConfig(level=5)`, which is even below DEBUG.

This also places it in a convenient spot where just enabling Python logging doesn't come on unintentionally, and pyo3_log's caching means that there should not be a practically relevant performance impact.

Comment (geonnave) [reply lakers-python/src/lib.rs:76]: Thank you, I just tested and it works as you said.

(previously I had misunderstood, and thought that traces logs would _always_ show even when python logs are off, but I verified that this is not the case)

**Comment by geonnave** [comment]
Thanks for this PR! Merging

**Pull request merged**